packages/aws_common/lib/src/http/aws_http_client_io.dart

```diff
@@ -0,0 +1,522 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:aws_common/aws_common.dart';
+import 'package:http2/http2.dart';
+import 'package:meta/meta.dart';
+import 'package:stream_transform/stream_transform.dart';
+
+/// {@template aws_common.http.http_client_impl}
+/// The platform-specific implementation of [AWSHttpClient].
+///
+/// - On the VM, this wraps `dart:io`'s [HttpClient] and `package:http2`.
+/// - On the Web, this uses the browser's `fetch` API.
+/// {@endtemplate}
+class AWSHttpClientImpl extends AWSHttpClient {
+  /// {@macro aws_common.http.http_client_impl}
+  AWSHttpClientImpl() : super.protected();
+
+  /// The underlying `dart:io` HTTP/1.1 client.
+  HttpClient? _inner;
+
+  final _http2Connections = <String, ClientTransportConnection>{};
+
+  static final AWSLogger _logger = AWSLogger().createChild('AWSHttpClient');
+
+  void _setBadCertificateCallback(
+    HttpClient client,
+    BadCertificateCallback callback,
+  ) {
+    client.badCertificateCallback = (cert, host, port) {
+      final certificate = X509Certificate(
```

Comment:
Converts the `dart:io` type to a local version which can be safely exported from the package.

Reply:
Ah I see. Maybe worth a comment. Or an extension on the other type to convert it since this is done in a couple locations. `final certificate = cert.toInternalCert()` or something.